### PR TITLE
Make PageTemplateDialog gtk4-compatible

### PR DESCRIPTION
**Pull request by bhennion**
This concludes the series of PR's about PageType-related widgetry. Based on #4508

- [x] Merge #4508 and rebase

### CODE REVIEW

File: src/core/gui/dialog/PageTemplateDialog.cpp [hunk 1]
```diff
@@ -13,40 +13,63 @@
...
-    g_signal_connect(get("btLoad"), "clicked",
-                     G_CALLBACK(+[](GtkToggleButton* togglebutton, PageTemplateDialog* self) { self->loadFromFile(); }),
-                     this);
+    g_signal_connect_swapped(builder.get("btSave"), "clicked",
+                             G_CALLBACK(+[](PageTemplateDialog* self) { self->saveToFile(); }), this);
 
-    g_signal_connect(get("btSave"), "clicked",
-                     G_CALLBACK(+[](GtkToggleButton* togglebutton, PageTemplateDialog* self) { self->saveToFile(); }),
-                     this);
+    g_signal_connect_swapped(builder.get("btCancel"), "clicked", G_CALLBACK(gtk_window_close), this->window.get());
```
Comment (rolandlo) [inline src/core/gui/dialog/PageTemplateDialog.cpp:65]: It looks more consistent to use the `getWindow()` method.
```suggestion
    g_signal_connect_swapped(builder.get("btCancel"), "clicked", G_CALLBACK(gtk_window_close), this->getWindow());
```

File: src/core/gui/dialog/PageTemplateDialog.cpp [hunk 2]
```diff
@@ -13,40 +13,63 @@
...
+    g_signal_connect_swapped(builder.get("btOk"), "clicked", G_CALLBACK(+[](PageTemplateDialog* self) {
+                                 self->saveToModel();
+                                 self->settings->setPageTemplate(self->model.toString());
+                                 self->toolMenuHandler->setDefaultNewPageType(self->model.getPageInsertType());
+                                 gtk_window_close(self->window.get());
```
Comment (rolandlo) [inline src/core/gui/dialog/PageTemplateDialog.cpp:70]: Same here:
```suggestion
                                 gtk_window_close(self->getWindow());
```

File: ui/pageTemplate.glade [hunk 3]
```diff
@@ -1,308 +1,301 @@
...
           </packing>
         </child>
       </object>
     </child>
-    <action-widgets>
-      <action-widget response="2">btCancel</action-widget>
-      <action-widget response="1">btOk</action-widget>
-    </action-widgets>
   </object>
 </interface>
```
Comment (rolandlo) [inline ui/pageTemplate.glade:301]: The GtkBox containing the buttons and the GtkBox containing everything don't have an ID. Not sure if that could ever be an issue, but as I recall everything in our Glade files had an ID so far. So maybe add one here, too.

**Review by rolandlo — COMMENTED** [review]
It works fine according to my tests. I have only some minor nitpicks about the code.

File: ui/pageTemplate.glade (hunk 3 above)
Comment (bhennion) [reply ui/pageTemplate.glade:301]: I'm not sure about that. I think that naming everything may be a mistake actually: I haven't looked under the hood, but I'd assume that GtkBuilder has a map id->GObject* of named objects. Uselessly giving widgets an id would just increase the map's size thus slowing down (even so slightly) the name-matching searches.
It's probably not worth removing the unused id, but why not avoid adding more?

Comment (rolandlo) [reply ui/pageTemplate.glade:301]: Your arguments make sense to me. So let's avoid adding unused id's in future.

**Review by rolandlo — APPROVED** [review]

**Comment by bhennion** [comment]
Thanks for the review! Merging in 24 hours.

**Pull request merged**